segmentation_labeling_app/transforms/transformations.py

```diff
@@ -0,0 +1,225 @@
+
+    return transformed_x, transformed_y
+
+
+def get_centered_coordinate_box_video(coordinate_pair: Tuple[int, int],
+                                      box_size: Tuple[int, int],
+                                      video_array: np.ndarray):
+    """
+    Function to get video subset centered around coordinate pair with
+    size specified by box size tuple. Function takes a subset of each frame
+    and stacks together to get final video.
+    Args:
+        coordinate_pair: the coordinate on which to center the subset
+        box_size: the size of the subset box
+        video_array: the video to take the subset from array should be of
+                     form (t, w, h)
+
+    Returns:
+
+    """
+    transformed_coordinates = get_transformed_center(coordinate_pair, box_size,
+                                                     video_array.shape)
+    left_column = transformed_coordinates[0] - math.ceil(box_size[0] / 2)
+    right_column = transformed_coordinates[0] + math.ceil(box_size[0] / 2)
+    up_row = transformed_coordinates[1] - math.ceil(box_size[1] / 2)
+    down_row = transformed_coordinates[1] + math.ceil(box_size[1] / 2)
+    transformed_video = np.zeros(shape=(video_array.shape[0], box_size[0], box_size[1]))
+    for i, frame in enumerate(video_array):
+        transformed_video[i] = frame[up_row:down_row, left_column:right_column]
+    return transformed_video
+
+
+def generate_max_ave_proj_image(video: np.ndarray,
+                                projection_type: str = 'average'):
+    """
+    Returns a maximum projection or an average projection of a video in
+    numpy array format
+    Args:
+        video: The video to generate the projection with shape (t, w, h)
+        projection_type: maximum or average, what type of projection
+
+    Returns: A numpy array generated with the specified strategy
+
+    """
+    if projection_type == 'average':
+        return np.mean(video)
+    elif projection_type == 'maximum':
+        return np.max(video, axis=0)
+
+
+def normalize_video(video: np.ndarray):
+    """
+    Function to normalize video by its global max and maximum 8 bit value
+    Args:
+        video: Video to be normalized with shape (t, w, h)
+    Returns:
+        norm_frames: normalized video
+    """
+    return np.uint8(video / video.max() * 255)
+
+
+def transform_to_mp4(video: np.ndarray, output_path: str):
+    """
+    Function to transform 2p gray scale video into a mp4
+    video using imageio_ffmpeg.
+    Args:
+        video: Video to be transformed with shape (t, w, h)
+        output_path: Output path for the transformed video
+
+    Returns:
+
+    """
+    norm_video = normalize_video(video)
+    writer = mpg.write_frames(output_path,
+                              video[0].shape,
+                              pix_fmt_in="gray",
+                              pix_fmt_out="gray",
+                              fps=30)
```

Comment:
This should be based on the downsampling rate. If we downsample to 4Hz/fps) and then set it to 30 here it will be quite fast. Maybe then just add it as an arg in your function?

Reply:
That's my mistake I thought I had it as a variable.

Reply:
it is still hard-coded to 30. You want `fps=fps`

Reply:
thanks for finding that, I feel really dumb leaving that in hah.